yellow-paper/docs/public-vm/avm-circuit.md

```diff
@@ -4,6 +4,27 @@ sidebar_position: 1
 
 # AVM Circuit
 
+## Call pointer
+Each message call processed within a single VM circuit execution is assigned a unique **call pointer**. There is certain information that must be tracked by the VM circuit on a per-call basis. For example, each call will correspond to the execution of a different contract's bytecode, and each call will access call-specific memory. As a per-call unique identifier, call pointer enables bytecode and memory lookups, among other things, on a per-call basis.
+
+Call pointers are assigned based on execution order. A request's initial message call is assigned call pointer of `1`. The first nested message call encountered during execution is assigned call pointer of `1`. The VM circuit tracks the highest call pointer assigned thus far, and whenever a nested call instruction is encountered, it increments that value and assigns the result to that call.
```

Comment:
Whether you set '0' for the initial call pointer or '2' for the nested one, no? Currently they are both '1'.

Reply:
Good catch. Nested call should be 2.